LiteCore/tests/VectorQueryTest.cc

```diff
@@ -246,6 +246,66 @@ N_WAY_TEST_CASE_METHOD(SIFTVectorQueryTest, "Hybrid Vector Query", "[Query][.Vec
                          {85776, 90431, 92142, 92629, 94598, 94989, 104787, 106750, 113260, 116129});
 }
 
+N_WAY_TEST_CASE_METHOD(SIFTVectorQueryTest, "Hybrid vs plain Vector Query", "[Query][.VectorSearch]") {
+    readVectorDocs();
+    {
+        // Add some docs without vector data, to ensure that doesn't break indexing:
+        ExclusiveTransaction t(db);
+        writeMultipleTypeDocs(t);
+        t.commit();
+    }
+    createVectorIndex();
+
+    // Pre-condition: 100 documents are selected by "META().sequence % 100 = 0"
+    string                    queryStr = "SELECT count(*) FROM "s + collectionName + " WHERE META().sequence % 100 = 0";
+    Retained<Query>           query    = store->compileQuery(queryStr, QueryLanguage::kN1QL);
+    Retained<QueryEnumerator> e        = query->createEnumerator();
+    REQUIRE(e->next());
+    auto countBySecondaryWhere = e->columns()[0]->asInt();
+    REQUIRE(countBySecondaryWhere == 100);
+
+    // Plain query fails to select docs because of the additional WHERE term.
+    // ===================
+
+    queryStr = "SELECT META().id, VECTOR_DISTANCE(vecIndex) AS distance FROM "s + collectionName
+               + " WHERE VECTOR_MATCH(vecIndex, $target, 10) AND META().sequence % 100 = 0"
+                 " ORDER BY distance";
+    query                                         = store->compileQuery(queryStr, QueryLanguage::kN1QL);
+    Query::Options                        options = optionsWithTargetVector(kTargetVector, kData);
+    std::vector<std::pair<string, float>> result  = collectResults(query->createEnumerator(&options));
+
+    // This is plain query because of explicit max_results.
+    // VECTOR_MATCH finds first 10 docs without regard to the secondary WHERE term, META().sequence % 100 = 0,
+    // and they are filtered out by that predicate.
+    // Note: hybrid query gives the desired result, c.f. test "Hybrid Vector Query",
+    CHECK(result.size() == 0);
+
+    // Hybrid vectorsearch consistentence: as we increase the LIMIT, the results are consistent.
+    // =====================
+
+    queryStr = "SELECT META().id, VECTOR_DISTANCE(vecIndex) AS distance FROM "s + collectionName
+               + " WHERE VECTOR_MATCH(vecIndex, $target) AND META().sequence % 100 = 0"
+                 " ORDER BY distance";
+    query  = store->compileQuery(queryStr, QueryLanguage::kN1QL);
+    result = collectResults(query->createEnumerator(&options));
+
+    REQUIRE(result.size() > 10);
```

Comment:
This looks identical to the test "Hybrid Vector Query", except that this one doesn't verify the results.

Reply:
I basically take it as a truth. The purpose is to check the LIMIT in the hybrid query is consistent, exactly because of the test "Hybrid Vector Query."